snapshot-jms-test/src/main/java/com/hazelcast/jet/tests/snapshot/jms/VerificationProcessor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.snapshot.jms;
+
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.BroadcastKey;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.logging.ILogger;
+import java.util.Map;
+import java.util.PriorityQueue;
+
+import static com.hazelcast.jet.core.ProcessorMetaSupplier.preferLocalParallelismOne;
+
+public class VerificationProcessor extends AbstractProcessor {
+
+    public static final String CONSUMED_MESSAGES_MAP_NAME = "JmsSourceTest_latestCounter";
+    private static final int QUEUE_SIZE_LIMIT = 5_000;
+    private static final int PRINT_LOG_ITEMS = 5_000;
+
+    private final String name;
+
+    private boolean processed;
+    private long counter;
+    private final PriorityQueue<Long> queue = new PriorityQueue<>();
+    private ILogger logger;
+    private Map<String, Long> map;
```

Comment:
Would be nice to have this as `IMap`, it's a lot different thing than just `Map`. Took me a while to notice that it's an imap actually.